### PR TITLE
add support for onchain rhs

### PR DESCRIPTION
**Pull request by FacuDeLorenzo**
Centralized configuration for faster testing
Added Iden3OnchainSmtCredentialStatusPublisher

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -712,7 +681,7 @@ async function handleAuthRequestWithProfilesV3CircuitBeta() {
   };
 
   console.log('=================  credential auth request ===================');
-  const verifierDID = 'did:polygonid:polygon:mumbai:2qLWqgjWa1cGnmPwCreXuPQrfLrRrzDL1evD6AG7p7';
+  const verifierDID = 'did:opid:optimism:sepolia:yqNTkU2Qrtvn4WFAGkpyiNZz6MMYz9aSC2MdGKjeNA';
```
Comment (juandahl) [inline index.ts]: should be an env variable?

File: walletSetup.ts [hunk 2]
```diff
@@ -43,29 +43,32 @@ import {
...
-  conf.contractAddress = contractAddress;
-  conf.url = rpcUrl;
-
+import { ethers } from 'ethers';
+
+const circuitsFolder = process.env.CIRCUITS_PATH ?? '';
+const mongoDbConnection = process.env.MONGO_DB_CONNECTION ?? '';
+const rpcUrl = process.env.RPC_URL ?? '';
+const chainId = process.env.RHS_CHAIN_ID ? +process.env.RHS_CHAIN_ID : undefined;
+const StateContractAddress = process.env.CONTRACT_ADDRESS ?? '';
```
Comment (juandahl) [inline walletSetup.ts]: a config file should be useful maybe

File: walletSetup.ts [hunk 3]
```diff
@@ -78,31 +81,21 @@ export function initInMemoryDataStorage({
...
     url = mongodb.getUri();
   }
   const client = new MongoClient(url);
   await client.connect();
-  const db: Db = client.db('mongodb-sdk-example');
-
-  const conf: EthConnectionConfig = defaultEthConnectionConfig;
-  conf.contractAddress = contractAddress;
-  conf.url = rpcUrl;
+  const db: Db = client.db('mongodb- sdk-example');
```
Comment (juandahl) [inline walletSetup.ts]: move the name to a constant?

**Review by juandahl — APPROVED** [review]
LGTM

File: index.ts [hunk 4]
```diff
@@ -33,33 +33,15 @@ const rhsUrl = process.env.RHS_URL as string;
...
-  network: 'sepolia',
-  networkFlag: 0b1000_0000 | 0b0000_0010
-});
-core.registerDidMethodNetwork({
-  method: OPID_METHOD,
-  blockchain: 'optimism',
-  chainId: 10,
-  network: 'main',
-  networkFlag: 0b1000_0000 | 0b0000_0001
-});
```
Comment (matzapata) [inline index.ts]: Totally! But don't we need our updated package? I was thinking we can already install it using the GitHub repo URL. Let's add it in this same pr so we keep a functional main branch

File: walletSetup.ts [hunk 5]
```diff
@@ -43,29 +43,32 @@ import {
...
-  const conf: EthConnectionConfig = defaultEthConnectionConfig;
-  conf.contractAddress = contractAddress;
-  conf.url = rpcUrl;
-
+import { ethers } from 'ethers';
+
+const circuitsFolder = process.env.CIRCUITS_PATH ?? '';
+const mongoDbConnection = process.env.MONGO_DB_CONNECTION ?? '';
+const rpcUrl = process.env.RPC_URL ?? '';
+const chainId = process.env.RHS_CHAIN_ID ? +process.env.RHS_CHAIN_ID : undefined;
```
Comment (matzapata) [inline walletSetup.ts]: Doesn't seem like this is doing anything...

File: walletSetup.ts (hunk 2 above)
Comment (matzapata) [reply walletSetup.ts]: Agree here, maybe even cool to show the 2 possible configs

```
const onchainRhsConfig = {}
const offchainRhsConfig = {}

export { onchainRhsConfig as config }
```

or smth like this

**Review by matzapata — COMMENTED** [review]
Looks good!! just some comments ))

File: walletSetup.ts (hunk 3 above)
Comment (FacuDeLorenzo) [reply walletSetup.ts]: Sure! I'll actually move it to the env file

File: index.ts (hunk 1 above)
Comment (FacuDeLorenzo) [reply index.ts]: Yup, better as an env variable!

File: walletSetup.ts (hunk 2 above)
Comment (FacuDeLorenzo) [reply walletSetup.ts]: Awesome, including both ideas :) ty!

File: walletSetup.ts (hunk 5 above)
Comment (FacuDeLorenzo) [reply walletSetup.ts]: So it got replaced in favour of the config file!

File: index.ts (hunk 4 above)
Comment (FacuDeLorenzo) [reply index.ts]: Yas! Now we have opid-sdk as an npm package :)

**Pull request merged**